### ISSUE TITLE
PlatformCompatibliityAnalyzer does not like OSPlatform declared as a member

### ISSUE THREAD

**Issue by pranavkm** [dotnet/roslyn-analyzers#4189]
### Analyzer

CA1416	'Unsupported' is unsupported on 'windows'	

### Analyzer source

**SDK**: 5.0.100-rc.2.20466.16

### Describe the bug

The platform compatibility analyzer allows using `OSPlatform` as a variable, but not as class level member.

### Steps To Reproduce

```C#
    [UnsupportedOSPlatform("windows")]
    class SomeUnsupportedAPI
    {
    }

    class Program
    {
        public static readonly OSPlatform WindowsOSPlatform = OSPlatform.Create("windows");

        static void Main(string[] args)
        {
            var osPlatform = OSPlatform.Create("windows");
            if (!RuntimeInformation.IsOSPlatform(osPlatform))
            {
                new SomeUnsupportedAPI();
            }

            if (!RuntimeInformation.IsOSPlatform(WindowsOSPlatform))
            {
                new SomeUnsupportedAPI();
            }
        }
    }
```

### Expected behavior

The analyzer shouldn't warn since the two checks are equivalent.

### Actual behavior

The check that uses the static member prints a warning. Also happens if the member is non-static.

**Comment by mavasani** [comment]
I think this was one of the cases we had decided does not need to be handled in the original flow analysis implementation.

@buyaa-n @terrajobst @jeffhandley - I believe this can be handled, but one can certainly come up with more cases where it needs richer flow analysis across symbols to complete flow data values.

**Comment by buyaa-n** [comment]
Yeah, I would assume flow analysis for this would be much more expensive, especially for big/fat classes

**Comment by mavasani** [comment]
@buyaa-n feel free to close it out if you feel this is by design.

**Comment by jeffhandley** [comment]
I'm closing this as a known limitation and we can see if we receive more reports of this or similar scenarios.
